### DISCUSSION TITLE
Expo firebase messaging

### DISCUSSION THREAD

**Discussion by diegonicolas250** [invertase/react-native-firebase#5779] [Q&A] [answered]
Hey guys, i'm struggling with this, I need to use firebase/messaging, all I need to do is use the onMessage() function from messaging, is it possible using Expo?

**Comment by mikehardy** [comment, marked as answer]
How did it go when you tried it?
The instructions here should serve: https://rnfirebase.io/#expo

You won't need perf or crashlytics config plugins, and I don't believe the messaging module needs a plugin, so it should work

If you want or need more advanced items like notification icon changes and such, you'll need to add + configure a firebase.json file but I'm not sure how that goes with Expo.

Mostly, follow those instructions above and let us know how it goes.

**Reply by diegonicolas250** [reply]
Thanks for your response, it worked using the config plugins as explained in your link! 
```
 {
  "expo": {
    "android": {
      "googleServicesFile": "./google-services.json"
    },
    "ios": {
      "googleServicesFile": "./GoogleService-Info.plist"
    },
    "plugins": [
      "@react-native-firebase/app",
    ]
  }
}
```
